src/data/dfinsupp/basic.lean

```diff
@@ -1929,3 +1935,31 @@ add_monoid_hom.congr_fun (comp_lift_add_hom h.to_add_monoid_hom g) f
 end add_equiv
 
 end
+
+section finite_infinite
+
+instance dfinsupp.fintype {ι : Sort*} {π : ι → Sort*} [decidable_eq ι] [Π i, has_zero (π i)]
+  [fintype ι] [∀ i, fintype (π i)] :
+  fintype (Π₀ i, π i) :=
+fintype.of_equiv (Π i, π i) dfinsupp.equiv_fun_on_fintype.symm
+
+/-- Weaker form of `dfinsupp.fintype`. -/
```

Comment:
I think it's not exactly "weaker", just that you get weaker conclusion from weaker assumptions.